web/assets/js/GoogleAnalyticsService.js

```diff
@@ -0,0 +1,77 @@
+class GoogleAnalyticsService {
+
+    constructor() {
+        this.gaDataAttr = 'data-ga-action';
+        this.gaCategoryAttr = 'data-ga-category';
+        this.gaLabelAttr = 'data-ga-label';
+    }
+
+    init() {
+        const gaSelectors = document.querySelectorAll(`[${this.gaDataAttr}]`);
+
+        if (gaSelectors.length > 0) {
```

Comment:
You don't have to check anything here. If an array is empty, then it won't iterate.